packages/trader/src/App/Components/Elements/VideoPlayer/video-controls.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Text, Icon } from '@deriv/components';
+import { formatDurationTime } from '@deriv/shared';
+import VolumeControl from './volume-control';
+import PlaybackRateControl from './playback-rate-control';
+
+type TVideoControls = {
+    block_controls?: boolean;
+    current_time?: number;
+    dragStartHandler: (e: React.MouseEvent<HTMLSpanElement> | React.TouchEvent<HTMLSpanElement>) => void;
+    has_enlarged_dot?: boolean;
+    is_animated?: boolean;
+    is_ended?: boolean;
+    is_playing?: boolean;
+    is_mobile?: boolean;
+    is_muted?: boolean;
+    onRewind: (e: React.MouseEvent<HTMLDivElement>) => void;
+    onVolumeChange: (new_value: number) => void;
+    onPlaybackRateChange: (new_value: number) => void;
+    progress_bar_filled_ref: React.RefObject<HTMLDivElement>;
+    progress_bar_ref: React.RefObject<HTMLDivElement>;
+    progress_dot_ref: React.RefObject<HTMLSpanElement>;
+    show_controls?: boolean;
+    togglePlay: (e: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement>) => void;
+    toggleMute: (new_value: boolean) => void;
+    video_duration?: number;
+    volume?: number;
+};
+
+const VideoControls = ({
+    block_controls,
+    current_time,
+    dragStartHandler,
+    has_enlarged_dot,
+    is_animated,
+    is_ended,
+    is_playing,
+    is_mobile,
+    is_muted,
+    onRewind,
+    onVolumeChange,
+    onPlaybackRateChange,
+    progress_bar_filled_ref,
+    progress_bar_ref,
+    progress_dot_ref,
+    show_controls,
+    togglePlay,
+    toggleMute,
+    video_duration,
+    volume,
```

Comment:
a lot of props are not used inside VideoControls but only passed down to children, e.g. VolumeControl. Do you think we could spread some props here `...props` ?

Reply:
Sorry, but actually `show_controls,onRewind,progress_bar_ref,is_animated,is_ended,progress_bar_filled_ref,is_mobile,has_enlarged_dot,dragStartHandler,progress_dot_ref,block_controls,togglePlay` and `is_playing` are used in the component itself. So I would keep them as this